src/hooks/useTimer.tsx

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { useEffect, useState } from 'react';
+
+import { dateToTimestamp } from '../utils/dateUtils';
+
+type UseTimerReturn = {
+	hours: string;
+	minutes: string;
+	seconds: string;
+};
+
+const useTimer = (date: string | number | Date): UseTimerReturn => {
+	const [timespan, setTimespan] = useState(() => {
+		const difference = Date.now() - dateToTimestamp(date);
```

Comment:
Here declare outside the function used inside the useState hook so it's much cleaner and readable